### ISSUE TITLE
ubuntu new install of 8.0.4 - pdfviewer does not load

### ISSUE THREAD

**Issue by aptalca** [owncloud/core#17424]
Hi,

I just did a manual install of owncloud 8.0.4 on ubuntu server with nginx and mariadb and I cannot get pdfviewer to work.

Here is what I installed first on minimal ubuntu (with no LAMP prior): php5-cli php5-common php5-gd php5-pgsql php5-sqlite php5-mysqlnd php5-curl php5-intl php5-mcrypt php5-ldap php5-gmp php5-imagick php5-fpm php5-gd smbclient nginx openssl wget bzip2 mariadb-server libreoffice

Everything seems to work fine except for the pdfviewer

When I click on a pdf, I get a blank window with just the navigation bar at the top. I tried chrome, firefox and ie so it's not my browser. I tried a demo owncloud server and the pdfviewer loaded. So there is definitely something wrong with my server setup.

When I try to open pdf's the logs don't show anything (owncloud.log in the data folder). I tried it with and without server side encryption, same result. I am able to download the pdfs, that works fine. I just can't view them within the browser.

The documents app works well for both viewing and editing word documents (through libreoffice), not sure if that matters.

Am I missing a dependency for pdf.js? Any ideas on how I can troubleshoot?

Thanks

EDIT: I just tried demo.owncloud.org again. I guess it was updated to 8.0.4 as well and the pdfviewer does not seem to work there either. When I tried the other day, it had worked.

**Comment by RealRancor** [comment]
Hi,

did you had a look at your browsers console as described in:

https://github.com/owncloud/core/blob/master/CONTRIBUTING.md#short-version
(Which should be used anyway for new issues)

The demo.owncloud.org was updated to 8.1 so this seems to be related.

**Comment by aptalca** [comment]
RealCancor,

Thanks so much for steering me in the right direction.

I checked the javascript console and saw that my browser was blocking all x-frames on the domain.

A quick google search revealed I had to add the option to allow same origin x-frames in my nginx config on the server. Now I can view pdf's.

I checked the javascript console on the demo server and it is the same exact error. The admin of the demo server needs to allow same origin x-frames.

For reference, I added the following line to my nginx.conf file:

`add_header X-Frame-Options "SAMEORIGIN";`

Thanks so much
